hbase-server/src/main/java/org/apache/hadoop/hbase/master/ServerManager.java

```diff
@@ -620,8 +620,9 @@ public synchronized boolean expireServer(final ServerName serverName) {
     }
   }
 
+  /** This is already called from a synchronized method, so synch is for visibility sake. */
   @VisibleForTesting
-  public void moveFromOnlineToDeadServers(final ServerName sn) {
+  public synchronized void moveFromOnlineToDeadServers(final ServerName sn) {
```

Comment:
Please remove the duplicated synchronized modifier, as there is an VisibleForTesting annotation which means this method will only be called inside the class, and we have synchronized in side the method, which will make others confusing?

Reply:
I'll rename the method instead to make it obvious it requires sync, in case someone calls it from other parts of this class

Reply:
Heh, actually it's not visible for testing at all. It's used in RPC... so it was a bug.
I'll re-add synchrnoized back and remove VisibleForTesting